test/FunctionalTests/Balancer/DnsResolverTests.cs

```diff
@@ -188,10 +195,75 @@ public async Task RefreshAsync_MultipleCallsThenCancellation_CallCanceled()
             Assert.NotNull(result);
             Assert.Null(result!.Addresses);
             Assert.AreEqual(StatusCode.Unavailable, result!.Status.StatusCode);
-            Assert.AreEqual("Error getting DNS hosts for address 'dns:///localhost'. TaskCanceledException: A task was canceled.", result!.Status.Detail);
+            Assert.AreEqual("Error getting DNS hosts for address 'localhost'. TaskCanceledException: A task was canceled.", result!.Status.Detail);
             Assert.AreEqual("A task was canceled.", result!.Status.DebugException.Message);
         }
 
+        [Test]
+        public async Task DNS_Port_Works()
+        {
+            const int defaultPort = 1234; // Will be ignored because port is specified in address
+
+            ResolverResult? result = null;
+            var dnsResolver = CreateDnsResolver(new Uri("dns:///localhost:8080"), defaultPort);
+            dnsResolver.Start(r =>
+            {
+                result = r;
+            });
+
+            // Act
+            await dnsResolver.RefreshAsync(CancellationToken.None);
+
+            // Assert
+            Assert.False(HasLogException((ex) =>
+            {
+                return ex is SocketException;
+            }));
+
+            Assert.AreEqual(StatusCode.OK, result!.Status.StatusCode);
+
+            var addresses = result!.Addresses!;
+            Logger.LogInformation($"Got {addresses.Count} addresses.");
+
+            // Note: addresses returned from localhost depend on operating system.
+            // Validate whatever we get back has the specified port.
+            for (var i = 0; i < addresses.Count; i++)
+            {
+                var address = addresses[i];
+
+                Logger.LogInformation($"Address: {address}");
+                Assert.AreEqual(8080, address.EndPoint.Port);
+            }
+        }
+
+        [Test]
+        public async Task DefaultPort_Localhost_AddressesHaveDefaultPort()
+        {
+            ResolverResult? result = null;
```

Comment:
```suggestion
            ResolverResult? result = null;
            var defaultPort = 8081;
```

Nit: for use in test setup and assertions.